File: apple/Sources/ReactNativeWatchOS/Modifiers/LineLimit.swift
import SwiftUI

/// SwiftUI `.lineLimit(_:)`. Caps the number of lines text within the view
/// can use before truncating. An absent `value` leaves the view unchanged.
enum RNWLineLimitModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("lineLimit") { view, params, _ in
            guard let limit = params.int("value") else { return view }
            return AnyView(view.lineLimit(limit))
        }
    }
}

File: src/modifiers/lineLimit.ts
import { createModifier } from './createModifier';

export interface LineLimitParams {
  /// Maximum number of lines text may occupy before truncating.
  value?: number;
}

/// SwiftUI `.lineLimit(_:)`. Caps the number of lines text within the
/// view can use. Pass a number directly or an object.
export function lineLimit(params?: LineLimitParams | number) {
  if (typeof params === 'number') {
    return createModifier('lineLimit', { value: params });
  }
  return createModifier('lineLimit', params);
}
